features/src/main/scala/com/salesforce/op/utils/spark/OpVectorColumnMetadata.scala

```diff
@@ -75,6 +79,9 @@ case class OpVectorColumnMetadata
   assert(parentFeatureName.length == parentFeatureType.length,
     s"must provide both type and name for every parent feature," +
       s" names: $parentFeatureName and types: $parentFeatureType do not have the same length")
+  assert(indicatorValue.isEmpty || descriptorValue.isEmpty, "cannot have both an indicatorValue and a," +
```

Comment:
`cannot have both an indicatorValue and descriptorValue.`

Reply:
Since we are making breaking changes anyways, I keep thinking that a base trait + two specific classes makes even more sense. This way we would avoid asserts like this ^
```scala
trait OpVectorColumnMetadata
(
  def parentFeatureName: Seq[String]
  def parentFeatureType: Seq[String]
  def grouping: Option[String]
  def index: Int
)
case class CategoricalVectorColumnMetadata(
  parentFeatureName: Seq[String],
  parentFeatureType: Seq[String],
  grouping: Option[String],
  indicatorValue: Option[String],
  index: Int = 0
) extends OpVectorColumnMetadata

case class ContinuousVectorColumnMetadata(
  parentFeatureName: Seq[String],
  parentFeatureType: Seq[String],
  grouping: Option[String],
  descriptorValue: Option[String]
  index: Int = 0
) extends OpVectorColumnMetadata
```

Reply:
this would break our internal usage

Reply:
I will make a separate ticket do do this as part of metadata restructuring